src/main/java/org/spongepowered/api/item/inventory/InventoryTransformations.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * This file is part of SpongeAPI, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.item.inventory;
+
+import org.spongepowered.api.item.inventory.entity.Hotbar;
+import org.spongepowered.api.item.inventory.entity.MainPlayerInventory;
+import org.spongepowered.api.util.generator.dummy.DummyObjectProvider;
+
+/**
+ * Commonly used {@link InventoryTransformation}s
+ */
+public class InventoryTransformations {
+
+    /**
+     * A transformation that puts the Hotbar of a player inventory before the main inventory.
+     *
+     * <p>This transformation is equivalent to calling {@link Inventory#union(Inventory)} on the query
+     * for {@link Hotbar} with the query for {@link MainPlayerInventory} as parameter.</p>
+     */
+    public static final InventoryTransformation PLAYER_MAIN_HOTBAR_FIRST = DummyObjectProvider.createFor(InventoryTransformation.class, "PLAYER_MAIN_HOTBAR_FIRST");
+
+    /**
+     * A transformations that reverses the slot order.
+     */
+    public static final InventoryTransformation REVERSE = DummyObjectProvider.createFor(InventoryTransformation.class, "REVERSE");
+
+    /**
+     * A transformation that returns the incoming inventory without changing anything.
+     */
+    public static final InventoryTransformation NO_OP = DummyObjectProvider.createFor(InventoryTransformation.class, "NO_OP");
```

Comment:
How is this useful?

Reply:
It's useful when you have a method that accepts an `InventoryTransformation` and want to have the option of doing nothing without creating a method that doesn't accept an `InventoryTransformation` (which would result in ugly code - `if(has tranformation) { method a } else { method b }`.